spec/unit/solr_tools_spec.rb

```diff
@@ -15,20 +15,165 @@
+      allow_any_instance_of(@harvester_class).to receive(:ping_solr).and_return(true)
+      expect_any_instance_of(@harvester_class).to receive(:ping_solr)
+      allow_any_instance_of(@harvester_class).to receive(:ping_source).and_return(false)
+      expect_any_instance_of(@harvester_class).to receive(:ping_source)
+
+      @cli.options = { data_center: %w(nsidc), die_on_failure: false, environment: 'dev' }
+      expect { @cli.ping }.to raise_error(SystemExit) do |error|
+        expect(error.status).to eq(SearchSolrTools::Errors::HarvestError::ERRCODE_SOURCE_PING)
+      end
+    end
+
+    it 'returns error if source is up but solr is not' do
+      allow_any_instance_of(@harvester_class).to receive(:ping_solr).and_return(false)
+      expect_any_instance_of(@harvester_class).to receive(:ping_solr)
+      allow_any_instance_of(@harvester_class).to receive(:ping_source).and_return(true)
+      expect_any_instance_of(@harvester_class).to receive(:ping_source)
+
+      @cli.options = { data_center: %w(nsidc), die_on_failure: false, environment: 'dev' }
+      expect { @cli.ping }.to raise_error(SystemExit) do |error|
+        expect(error.status).to eq(SearchSolrTools::Errors::HarvestError::ERRCODE_SOLR_PING)
+      end
+    end
+
+    it 'returns error if neither solr nor source are up' do
+      allow_any_instance_of(@harvester_class).to receive(:ping_solr).and_return(false)
+      expect_any_instance_of(@harvester_class).to receive(:ping_solr)
+      allow_any_instance_of(@harvester_class).to receive(:ping_source).and_return(false)
+      expect_any_instance_of(@harvester_class).to receive(:ping_source)
+
+      @cli.options = { data_center: %w(nsidc), die_on_failure: false, environment: 'dev' }
+      expect { @cli.ping }.to raise_error(SystemExit) do |error|
+        expect(error.status).to eq(SearchSolrTools::Errors::HarvestError::ERRCODE_SOLR_PING + SearchSolrTools::Errors::HarvestError::ERRCODE_SOURCE_PING)
+      end
+    end
+  end
+
   describe '#harvest' do
+    let (:ingest_ok) { SearchSolrTools::Helpers::HarvestStatus::INGEST_OK }
+    let (:ingest_invalid_doc) { SearchSolrTools::Helpers::HarvestStatus::INGEST_ERR_INVALID_DOC }
+    let (:ingest_solr_err) { SearchSolrTools::Helpers::HarvestStatus::INGEST_ERR_SOLR_ERROR }
+    let (:doc_result) { { num_docs: 3, add_docs: ['doc1', 'doc2', 'doc3'], failure_ids: [] } }
+
+    before(:each) do
+      @harvester_class = SearchSolrTools::Harvesters::NsidcJson
+    end
+
     it 'calls the selected harvester classes' do
       puts 'CLI' + @cli.harvester_map.to_s
-      [SearchSolrTools::Harvesters::Adc, SearchSolrTools::Harvesters::Echo].each do |harvester_class|
-        allow_any_instance_of(harvester_class).to receive(:harvest_and_delete).and_return(true)
-        expect_any_instance_of(harvester_class).to receive(:harvest_and_delete)
+      [SearchSolrTools::Harvesters::Adc, SearchSolrTools::Harvesters::Echo].each do |test_harvester_class|
```

Comment:
Shouldn't be using ADC and/or ECHO here; only expect to see tests against NSIDC harvester.